deps/buffer.lua

```diff
@@ -27,42 +27,74 @@ limitations under the License.
   tags = {"luvit", "buffer"}
 ]]
 
-if not pcall(require, 'ffi') then
-  error("The 'buffer' module requires FFI support, which is not available on this platform.")
-end
-
+local bit = require('bit')
 local core = require('core')
-local ffi = require('ffi')
 
+local char = string.char
+local band = bit.band
+local unpack = unpack or table.unpack
 local Object = core.Object
 local instanceof = core.instanceof
 
-ffi.cdef[[
-  void *malloc (size_t __size);
-  void *calloc (size_t nmemb, size_t __size);
-  void free (void *__ptr);
-]]
+local isWindows = require('los').type() == 'win32'
+local hasFfi, ffi = pcall(require, 'ffi')
+
+local C
+if hasFfi then
+  ffi.cdef[[
+    void *malloc (size_t __size);
+    void *calloc (size_t nmemb, size_t __size);
+    void free (void *__ptr);
+  ]]
+  -- avoid bugs when linked with static runtime libraries, eg. /MT link flags
+  C = isWindows and ffi.load("msvcrt") or ffi.C
+end
 
 local buffer = {}
 
 local Buffer = Object:extend()
 buffer.Buffer = Buffer
 
---avoid bugs when link with static run times lib, eg. /MT link flags
-local C = ffi.os == "Windows" and ffi.load("msvcrt") or ffi.C
+local function castChar(val)
+  if val >= 0 and val <= 0xff then
+    return val
+  else
+    return band(val, 0xff)
+  end
+end
```

Comment:
This function is unnecessary, `band(val, 0xff)` exhibits the desired behavior

Reply:
Correct, the if statement is meant as an early return to not unnecessarily AND the value. I will remove it though, seems to have a negative effect on performance rather than a good one